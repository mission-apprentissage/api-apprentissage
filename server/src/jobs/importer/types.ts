import type { CronDef } from "job-processor";
import type { ImportStatus } from "shared";

export type Importer = CronDef & {
  getStatus(): Promise<ImportStatus>;
};
