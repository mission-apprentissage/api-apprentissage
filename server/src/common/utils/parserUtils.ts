import { Options, parse } from "csv-parse";
import { isEmpty, pickBy } from "lodash-es";

export const DEFAULT_DELIMITER = ";";

export function parseCsv(options: Options = {}) {
  return parse({
    trim: true,
    delimiter: DEFAULT_DELIMITER,
    columns: true,
    bom: true,
    relax_column_count: true,
    on_record: (record) => {
      return pickBy(record, (v) => {
        return !isEmpty(v) && v.trim().length;
      });
    },
    ...options,
  });
}
