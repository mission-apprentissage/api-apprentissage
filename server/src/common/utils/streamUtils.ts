import { Transform } from "node:stream";

type Options = {
  size: number;
};

export function createBatchTransformStream(opts: Options): Transform {
  let currentBatch: unknown[] = [];

  return new Transform({
    objectMode: true,
    transform(chunk, _encoding, callback) {
      currentBatch.push(chunk);

      if (currentBatch.length >= opts.size) {
        this.push(currentBatch);
        currentBatch = [];
      }

      callback();
    },
    flush(callback) {
      if (currentBatch.length > 0) {
        this.push(currentBatch);
      }
      callback();
    },
  });
}
