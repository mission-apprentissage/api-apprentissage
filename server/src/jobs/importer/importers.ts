import { getAcceeImporterStatus, runAcceImporter } from "./acce/acce.js";
import { getBcnImporterStatus, runBcnImporter } from "./bcn/bcn.importer.js";
import { getCatalogueImporterStatus, runCatalogueImporter } from "./catalogue/catalogue.importer.js";
import { getCertificationImporterStatus, importCertifications } from "./certifications/certifications.importer.js";
import { getCommuneImporterStatus, runCommuneImporter } from "./commune/commune.importer.js";
import { getDaresApiIdccImporterStatus, runDaresApeIdccImporter } from "./dares/ape_idcc/dares.ape_idcc.importer.js";
import { getDaresCcnImporterStatus, runDaresConventionCollectivesImporter } from "./dares/ccn/dares.ccn.importer.js";
import { getFormationsImporterStatus, importFormations } from "./formation/formation.importer.js";
import { getFranceCompetencesImporterStatus, runRncpImporter } from "./france_competence/france_competence.importer.js";
import { getKaliImporterStatus, runKaliConventionCollectivesImporter } from "./kali/kali.ccn.importer.js";
import { getKitApprentissageImporterStatus, runKitApprentissageImporter } from "./kit/kitApprentissage.importer.js";
import { getMissionLocaleImporterStatus, runMissionLocaleImporter } from "./mission_locale/mission_locale.importer.js";
import { getOrganismesImporterStatus, importOrganismes } from "./organisme/organisme.importer.js";
import { getReferentielImporterStatus, runReferentielImporter } from "./referentiel/referentiel.js";
import type { Importer } from "./types.js";

const timings = {
  import_source: "0 4 * * *",
  import_source_main: "0 */4 * * *",
};

export const importers: Record<string, Importer> = {
  "Mise à jour acce": {
    cron_string: timings.import_source,
    handler: runAcceImporter,
    resumable: true,
    getStatus: getAcceeImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
  "Import des données BCN": {
    cron_string: timings.import_source_main,
    handler: runBcnImporter,
    resumable: true,
    getStatus: getBcnImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
  "Import des données Kit Apprentissage": {
    cron_string: timings.import_source_main,
    handler: runKitApprentissageImporter,
    resumable: true,
    getStatus: getKitApprentissageImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
  "Import des données Referentiel": {
    cron_string: timings.import_source_main,
    handler: runReferentielImporter,
    resumable: true,
    getStatus: getReferentielImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
  "Import des données Catalogue": {
    cron_string: timings.import_source_main,
    handler: runCatalogueImporter,
    resumable: true,
    getStatus: getCatalogueImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
  "Import des données France Compétences": {
    cron_string: timings.import_source,
    handler: runRncpImporter,
    resumable: true,
    getStatus: getFranceCompetencesImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
  "Import des certifications": {
    cron_string: timings.import_source_main,
    handler: async () => importCertifications(),
    resumable: true,
    getStatus: getCertificationImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
  "Import des organismes": {
    cron_string: timings.import_source_main,
    handler: async () => importOrganismes(false),
    resumable: true,
    getStatus: getOrganismesImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
  "Import des formations": {
    cron_string: timings.import_source_main,
    handler: importFormations,
    resumable: true,
    getStatus: getFormationsImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
  "Import des Conventions Collective Kali": {
    cron_string: timings.import_source,
    handler: runKaliConventionCollectivesImporter,
    resumable: true,
    getStatus: getKaliImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
  "Import des Conventions Collective Dares": {
    cron_string: timings.import_source,
    handler: runDaresConventionCollectivesImporter,
    resumable: true,
    getStatus: getDaresCcnImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
  "Import des APE-IDCC Dares": {
    cron_string: timings.import_source,
    handler: runDaresApeIdccImporter,
    resumable: true,
    getStatus: getDaresApiIdccImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
  "Import des Missions Locales": {
    cron_string: timings.import_source,
    handler: runMissionLocaleImporter,
    resumable: true,
    getStatus: getMissionLocaleImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
  "Import des Communes": {
    cron_string: timings.import_source,
    handler: runCommuneImporter,
    resumable: true,
    getStatus: getCommuneImporterStatus,
    checkinMargin: 60, // 1h
    maxRuntimeInMinutes: 30,
  },
};
