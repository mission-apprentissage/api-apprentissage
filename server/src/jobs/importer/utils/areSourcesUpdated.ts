export function areSourcesUpdated(
  lastImportSources: Record<string, { import_date: Date }> | null | undefined,
  currentSources: Record<string, { import_date: Date }>
): boolean {
  if (!lastImportSources) {
    return true;
  }

  for (const source in currentSources) {
    if (!lastImportSources[source] || lastImportSources[source].import_date < currentSources[source].import_date) {
      return true;
    }
  }

  return false;
}
