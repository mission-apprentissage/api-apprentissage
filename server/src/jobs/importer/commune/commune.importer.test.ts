import { ObjectId } from "mongodb";
import {
  academieFixtures,
  inseeAnciennesFixtures,
  inseeArrondissementFixtures,
  inseeCollectiviteFixtures,
  sourceCodeInseeToMissionLocaleFixture,
  sourceCommuneFixtures,
  sourceDepartementFixtures,
  sourceRegionExtendedFixtures,
  sourceRegionsFixtures,
} from "shared/models/fixtures/commune.model.fixture";
import { beforeEach, describe, expect, it, vi } from "vitest";
import { runCommuneImporter } from "./commune.importer.js";
import { useMongo } from "@tests/mongo.test.utils.js";

import { fetchAcademies } from "@/services/apis/enseignementSup/enseignementSup.js";
import { fetchGeoCommunes, fetchGeoDepartements, fetchGeoRegion, fetchGeoRegions } from "@/services/apis/geo/geo.js";
import {
  fetchAnciennesCommuneByCodeCommune,
  fetchArrondissementIndexedByCodeCommune,
  fetchCollectivitesOutreMer,
} from "@/services/apis/insee/insee.js";
import { getDbCollection } from "@/services/mongodb/mongodbService.js";

useMongo();

vi.mock("@/services/apis/geo/geo.js");
vi.mock("@/services/apis/enseignementSup/enseignementSup.js");
vi.mock("@/services/apis/insee/insee.js");

describe("runCommuneImporter", () => {
  const now = new Date("2024-10-03T21:53:08.141Z");
  const yesterday = new Date("2024-10-02T21:53:08.141Z");

  beforeEach(async () => {
    vi.useFakeTimers();
    vi.setSystemTime(now);

    const import_id = new ObjectId();
    await getDbCollection("source.insee_to_ml").insertMany(
      sourceCodeInseeToMissionLocaleFixture.map((d) => ({
        ...d,
        _id: new ObjectId(),
        import_id,
      }))
    );

    return () => vi.useRealTimers();
  });

  it("should import initial communes", async () => {
    vi.mocked(fetchGeoRegions).mockResolvedValue(sourceRegionsFixtures);
    vi.mocked(fetchGeoRegion).mockImplementation(
      async (code: string) => sourceRegionExtendedFixtures.find((r) => r.code === code)!
    );
    vi.mocked(fetchGeoDepartements).mockImplementation(
      async (codeRegion: string) => sourceDepartementFixtures[codeRegion as keyof typeof sourceDepartementFixtures]
    );
    vi.mocked(fetchGeoCommunes).mockImplementation(
      async (codeDepartement: string) => sourceCommuneFixtures[codeDepartement as keyof typeof sourceCommuneFixtures]
    );
    vi.mocked(fetchCollectivitesOutreMer).mockResolvedValue(inseeCollectiviteFixtures);
    vi.mocked(fetchAcademies).mockResolvedValue(academieFixtures);
    vi.mocked(fetchArrondissementIndexedByCodeCommune).mockResolvedValue(inseeArrondissementFixtures);
    vi.mocked(fetchAnciennesCommuneByCodeCommune).mockResolvedValue(inseeAnciennesFixtures);

    await runCommuneImporter();

    expect(fetchGeoRegions).toHaveBeenCalledTimes(1);
    expect(fetchGeoRegion).toHaveBeenCalledTimes(2);
    expect(fetchGeoDepartements).toHaveBeenCalledTimes(4);
    expect(fetchGeoCommunes).toHaveBeenCalledTimes(7);

    const communes = await getDbCollection("commune").find({}).toArray();

    expect(
      communes.map(({ _id, ...rest }) => rest).toSorted((a, b) => a.code.insee.localeCompare(b.code.insee))
    ).toMatchSnapshot();

    const importMetas = await getDbCollection("import.meta").find().toArray();

    expect(importMetas).toEqual([
      {
        _id: expect.any(ObjectId),
        import_date: now,
        type: "communes",
        status: "done",
      },
    ]);
  });

  it("should import communes with existing regions and departements", async () => {
    await getDbCollection("commune").insertMany(
      sourceCommuneFixtures["75"].map((geoCommune) => ({
        _id: new ObjectId(),
        nom: "Ancien nom",
        code: { insee: geoCommune.code, postaux: geoCommune.codesPostaux },
        departement: {
          nom: "Ancien nom departement",
          codeInsee: "Ancien code departement",
        },
        region: {
          codeInsee: "Ancien code region",
          nom: "Ancien nom region",
        },
        academie: {
          code: "Ancien code",
          nom: "Ancien nom",
          id: "Ancien id",
        },
        localisation: {
          centre: { type: "Point", coordinates: [0, 0] },
          bbox: {
            type: "Polygon",
            coordinates: [
              [
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
              ],
            ],
          },
        },
        mission_locale: null,
        arrondissements: [],
        anciennes: [],
        updated_at: yesterday,
        created_at: yesterday,
      }))
    );
    await getDbCollection("commune").insertOne({
      _id: new ObjectId(),
      nom: "Ancien nom",
      code: { insee: "Code supprimÃ©", postaux: [] },
      departement: {
        nom: "Ancien nom departement",
        codeInsee: "Ancien code departement",
      },
      region: {
        codeInsee: "Ancien code region",
        nom: "Ancien nom region",
      },
      academie: {
        code: "Ancien code",
        nom: "Ancien nom",
        id: "Ancien id",
      },
      localisation: {
        centre: { type: "Point", coordinates: [0, 0] },
        bbox: {
          type: "Polygon",
          coordinates: [
            [
              [0, 0],
              [0, 0],
              [0, 0],
              [0, 0],
            ],
          ],
        },
      },
      mission_locale: null,
      arrondissements: [],
      anciennes: [],
      updated_at: yesterday,
      created_at: yesterday,
    });
    await getDbCollection("import.meta").insertOne({
      _id: new ObjectId(),
      import_date: yesterday,
      type: "communes",
      status: "done",
    });

    vi.mocked(fetchGeoRegions).mockResolvedValue(sourceRegionsFixtures);
    vi.mocked(fetchGeoRegion).mockImplementation(
      async (code: string) => sourceRegionExtendedFixtures.find((r) => r.code === code)!
    );
    vi.mocked(fetchGeoDepartements).mockImplementation(
      async (codeRegion: string) => sourceDepartementFixtures[codeRegion as keyof typeof sourceDepartementFixtures]
    );
    vi.mocked(fetchGeoCommunes).mockImplementation(
      async (codeDepartement: string) => sourceCommuneFixtures[codeDepartement as keyof typeof sourceCommuneFixtures]
    );

    vi.mocked(fetchCollectivitesOutreMer).mockResolvedValue(inseeCollectiviteFixtures);
    vi.mocked(fetchAcademies).mockResolvedValue(academieFixtures);
    vi.mocked(fetchArrondissementIndexedByCodeCommune).mockResolvedValue(inseeArrondissementFixtures);
    vi.mocked(fetchAnciennesCommuneByCodeCommune).mockResolvedValue(inseeAnciennesFixtures);

    await runCommuneImporter();

    expect(fetchGeoRegions).toHaveBeenCalledTimes(1);
    expect(fetchGeoRegion).toHaveBeenCalledTimes(2);
    expect(fetchGeoDepartements).toHaveBeenCalledTimes(4);
    expect(fetchGeoCommunes).toHaveBeenCalledTimes(7);

    const communes = await getDbCollection("commune").find({}).toArray();

    expect(
      communes.map(({ _id, ...rest }) => rest).toSorted((a, b) => a.code.insee.localeCompare(b.code.insee))
    ).toMatchSnapshot();

    const importMetas = await getDbCollection("import.meta").find().toArray();

    expect(importMetas).toEqual([
      {
        _id: expect.any(ObjectId),
        import_date: yesterday,
        type: "communes",
        status: "done",
      },
      {
        _id: expect.any(ObjectId),
        import_date: now,
        type: "communes",
        status: "done",
      },
    ]);
  });
});
