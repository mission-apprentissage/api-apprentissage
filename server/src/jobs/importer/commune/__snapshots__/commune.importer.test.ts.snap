// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`runCommuneImporter > should import communes with existing regions and departements 1`] = `
[
  {
    "academie": {
      "code": "18",
      "id": "A18",
      "nom": "Orléans-Tours",
    },
    "code": {
      "insee": "18001",
      "postaux": [
        "18250",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "18",
      "nom": "Cher",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              2.437874,
              47.25828,
            ],
            [
              2.482241,
              47.25828,
            ],
            [
              2.482241,
              47.307371,
            ],
            [
              2.437874,
              47.307371,
            ],
            [
              2.437874,
              47.25828,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          2.4601,
          47.2828,
        ],
        "type": "Point",
      },
    },
    "nom": "Achères",
    "region": {
      "codeInsee": "24",
      "nom": "Centre-Val de Loire",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "18",
      "id": "A18",
      "nom": "Orléans-Tours",
    },
    "code": {
      "insee": "18002",
      "postaux": [
        "18200",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "18",
      "nom": "Cher",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              2.510546,
              46.625569,
            ],
            [
              2.572084,
              46.625569,
            ],
            [
              2.572084,
              46.689055,
            ],
            [
              2.510546,
              46.689055,
            ],
            [
              2.510546,
              46.625569,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          2.5413,
          46.6573,
        ],
        "type": "Point",
      },
    },
    "nom": "Ainay-le-Vieil",
    "region": {
      "codeInsee": "24",
      "nom": "Centre-Val de Loire",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "18",
      "id": "A18",
      "nom": "Orléans-Tours",
    },
    "code": {
      "insee": "41001",
      "postaux": [
        "41310",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "41",
      "nom": "Loir-et-Cher",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              0.940639,
              47.683749,
            ],
            [
              1.000203,
              47.683749,
            ],
            [
              1.000203,
              47.735078,
            ],
            [
              0.940639,
              47.735078,
            ],
            [
              0.940639,
              47.683749,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          0.9704,
          47.7094,
        ],
        "type": "Point",
      },
    },
    "nom": "Ambloy",
    "region": {
      "codeInsee": "24",
      "nom": "Centre-Val de Loire",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "18",
      "id": "A18",
      "nom": "Orléans-Tours",
    },
    "code": {
      "insee": "45001",
      "postaux": [
        "45230",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "45",
      "nom": "Loiret",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              2.749006,
              47.727449,
            ],
            [
              2.835944,
              47.727449,
            ],
            [
              2.835944,
              47.790032,
            ],
            [
              2.749006,
              47.790032,
            ],
            [
              2.749006,
              47.727449,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          2.7925,
          47.7587,
        ],
        "type": "Point",
      },
    },
    "nom": "Adon",
    "region": {
      "codeInsee": "24",
      "nom": "Centre-Val de Loire",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "01",
      "id": "A01",
      "nom": "Paris",
    },
    "code": {
      "insee": "75056",
      "postaux": [
        "75001",
        "75002",
        "75003",
        "75004",
        "75005",
        "75006",
        "75007",
        "75008",
        "75009",
        "75010",
        "75011",
        "75012",
        "75013",
        "75014",
        "75015",
        "75016",
        "75017",
        "75018",
        "75019",
        "75020",
        "75116",
      ],
    },
    "created_at": 2024-10-02T21:53:08.141Z,
    "departement": {
      "codeInsee": "75",
      "nom": "Paris",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              2.224219,
              48.815562,
            ],
            [
              2.469851,
              48.815562,
            ],
            [
              2.469851,
              48.902148,
            ],
            [
              2.224219,
              48.902148,
            ],
            [
              2.224219,
              48.815562,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          2.347,
          48.8589,
        ],
        "type": "Point",
      },
    },
    "nom": "Paris",
    "region": {
      "codeInsee": "11",
      "nom": "Île-de-France",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "24",
      "id": "A24",
      "nom": "Créteil",
    },
    "code": {
      "insee": "77001",
      "postaux": [
        "77760",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "77",
      "nom": "Seine-et-Marne",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              2.541179,
              48.323765,
            ],
            [
              2.589357,
              48.323765,
            ],
            [
              2.589357,
              48.371343,
            ],
            [
              2.541179,
              48.371343,
            ],
            [
              2.541179,
              48.323765,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          2.5653,
          48.3476,
        ],
        "type": "Point",
      },
    },
    "nom": "Achères-la-Forêt",
    "region": {
      "codeInsee": "11",
      "nom": "Île-de-France",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "24",
      "id": "A24",
      "nom": "Créteil",
    },
    "code": {
      "insee": "77002",
      "postaux": [
        "77120",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "77",
      "nom": "Seine-et-Marne",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              3.106556,
              48.702079,
            ],
            [
              3.173488,
              48.702079,
            ],
            [
              3.173488,
              48.763362,
            ],
            [
              3.106556,
              48.763362,
            ],
            [
              3.106556,
              48.702079,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          3.14,
          48.7327,
        ],
        "type": "Point",
      },
    },
    "nom": "Amillis",
    "region": {
      "codeInsee": "11",
      "nom": "Île-de-France",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "44",
      "id": "A44",
      "nom": "Saint-Pierre-et-Miquelon",
    },
    "code": {
      "insee": "98801",
      "postaux": [
        "98811",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "975",
      "nom": "Saint-Pierre-et-Miquelon",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              163.569697,
              -19.902093,
            ],
            [
              163.708819,
              -19.902093,
            ],
            [
              163.708819,
              -19.525114,
            ],
            [
              163.569697,
              -19.525114,
            ],
            [
              163.569697,
              -19.902093,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          163.6412,
          -19.711,
        ],
        "type": "Point",
      },
    },
    "nom": "Belep",
    "region": {
      "codeInsee": "975",
      "nom": "Saint-Pierre-et-Miquelon",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "41",
      "id": "A41",
      "nom": "Polynésie Française",
    },
    "code": {
      "insee": "98901",
      "postaux": [
        "98799",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "989",
      "nom": "Île de Clipperton",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              -109.234607,
              10.287154,
            ],
            [
              -109.19979,
              10.287154,
            ],
            [
              -109.19979,
              10.31957,
            ],
            [
              -109.234607,
              10.31957,
            ],
            [
              -109.234607,
              10.287154,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          -109.2172,
          10.3034,
        ],
        "type": "Point",
      },
    },
    "nom": "Île de Clipperton",
    "region": {
      "codeInsee": "989",
      "nom": "Île de Clipperton",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
]
`;

exports[`runCommuneImporter > should import initial communes 1`] = `
[
  {
    "academie": {
      "code": "18",
      "id": "A18",
      "nom": "Orléans-Tours",
    },
    "code": {
      "insee": "18001",
      "postaux": [
        "18250",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "18",
      "nom": "Cher",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              2.437874,
              47.25828,
            ],
            [
              2.482241,
              47.25828,
            ],
            [
              2.482241,
              47.307371,
            ],
            [
              2.437874,
              47.307371,
            ],
            [
              2.437874,
              47.25828,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          2.4601,
          47.2828,
        ],
        "type": "Point",
      },
    },
    "nom": "Achères",
    "region": {
      "codeInsee": "24",
      "nom": "Centre-Val de Loire",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "18",
      "id": "A18",
      "nom": "Orléans-Tours",
    },
    "code": {
      "insee": "18002",
      "postaux": [
        "18200",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "18",
      "nom": "Cher",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              2.510546,
              46.625569,
            ],
            [
              2.572084,
              46.625569,
            ],
            [
              2.572084,
              46.689055,
            ],
            [
              2.510546,
              46.689055,
            ],
            [
              2.510546,
              46.625569,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          2.5413,
          46.6573,
        ],
        "type": "Point",
      },
    },
    "nom": "Ainay-le-Vieil",
    "region": {
      "codeInsee": "24",
      "nom": "Centre-Val de Loire",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "18",
      "id": "A18",
      "nom": "Orléans-Tours",
    },
    "code": {
      "insee": "41001",
      "postaux": [
        "41310",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "41",
      "nom": "Loir-et-Cher",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              0.940639,
              47.683749,
            ],
            [
              1.000203,
              47.683749,
            ],
            [
              1.000203,
              47.735078,
            ],
            [
              0.940639,
              47.735078,
            ],
            [
              0.940639,
              47.683749,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          0.9704,
          47.7094,
        ],
        "type": "Point",
      },
    },
    "nom": "Ambloy",
    "region": {
      "codeInsee": "24",
      "nom": "Centre-Val de Loire",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "18",
      "id": "A18",
      "nom": "Orléans-Tours",
    },
    "code": {
      "insee": "45001",
      "postaux": [
        "45230",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "45",
      "nom": "Loiret",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              2.749006,
              47.727449,
            ],
            [
              2.835944,
              47.727449,
            ],
            [
              2.835944,
              47.790032,
            ],
            [
              2.749006,
              47.790032,
            ],
            [
              2.749006,
              47.727449,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          2.7925,
          47.7587,
        ],
        "type": "Point",
      },
    },
    "nom": "Adon",
    "region": {
      "codeInsee": "24",
      "nom": "Centre-Val de Loire",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "01",
      "id": "A01",
      "nom": "Paris",
    },
    "code": {
      "insee": "75056",
      "postaux": [
        "75001",
        "75002",
        "75003",
        "75004",
        "75005",
        "75006",
        "75007",
        "75008",
        "75009",
        "75010",
        "75011",
        "75012",
        "75013",
        "75014",
        "75015",
        "75016",
        "75017",
        "75018",
        "75019",
        "75020",
        "75116",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "75",
      "nom": "Paris",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              2.224219,
              48.815562,
            ],
            [
              2.469851,
              48.815562,
            ],
            [
              2.469851,
              48.902148,
            ],
            [
              2.224219,
              48.902148,
            ],
            [
              2.224219,
              48.815562,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          2.347,
          48.8589,
        ],
        "type": "Point",
      },
    },
    "nom": "Paris",
    "region": {
      "codeInsee": "11",
      "nom": "Île-de-France",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "24",
      "id": "A24",
      "nom": "Créteil",
    },
    "code": {
      "insee": "77001",
      "postaux": [
        "77760",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "77",
      "nom": "Seine-et-Marne",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              2.541179,
              48.323765,
            ],
            [
              2.589357,
              48.323765,
            ],
            [
              2.589357,
              48.371343,
            ],
            [
              2.541179,
              48.371343,
            ],
            [
              2.541179,
              48.323765,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          2.5653,
          48.3476,
        ],
        "type": "Point",
      },
    },
    "nom": "Achères-la-Forêt",
    "region": {
      "codeInsee": "11",
      "nom": "Île-de-France",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "24",
      "id": "A24",
      "nom": "Créteil",
    },
    "code": {
      "insee": "77002",
      "postaux": [
        "77120",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "77",
      "nom": "Seine-et-Marne",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              3.106556,
              48.702079,
            ],
            [
              3.173488,
              48.702079,
            ],
            [
              3.173488,
              48.763362,
            ],
            [
              3.106556,
              48.763362,
            ],
            [
              3.106556,
              48.702079,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          3.14,
          48.7327,
        ],
        "type": "Point",
      },
    },
    "nom": "Amillis",
    "region": {
      "codeInsee": "11",
      "nom": "Île-de-France",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "44",
      "id": "A44",
      "nom": "Saint-Pierre-et-Miquelon",
    },
    "code": {
      "insee": "98801",
      "postaux": [
        "98811",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "975",
      "nom": "Saint-Pierre-et-Miquelon",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              163.569697,
              -19.902093,
            ],
            [
              163.708819,
              -19.902093,
            ],
            [
              163.708819,
              -19.525114,
            ],
            [
              163.569697,
              -19.525114,
            ],
            [
              163.569697,
              -19.902093,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          163.6412,
          -19.711,
        ],
        "type": "Point",
      },
    },
    "nom": "Belep",
    "region": {
      "codeInsee": "975",
      "nom": "Saint-Pierre-et-Miquelon",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
  {
    "academie": {
      "code": "41",
      "id": "A41",
      "nom": "Polynésie Française",
    },
    "code": {
      "insee": "98901",
      "postaux": [
        "98799",
      ],
    },
    "created_at": 2024-10-03T21:53:08.141Z,
    "departement": {
      "codeInsee": "989",
      "nom": "Île de Clipperton",
    },
    "localisation": {
      "bbox": {
        "coordinates": [
          [
            [
              -109.234607,
              10.287154,
            ],
            [
              -109.19979,
              10.287154,
            ],
            [
              -109.19979,
              10.31957,
            ],
            [
              -109.234607,
              10.31957,
            ],
            [
              -109.234607,
              10.287154,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "centre": {
        "coordinates": [
          -109.2172,
          10.3034,
        ],
        "type": "Point",
      },
    },
    "nom": "Île de Clipperton",
    "region": {
      "codeInsee": "989",
      "nom": "Île de Clipperton",
    },
    "updated_at": 2024-10-03T21:53:08.141Z,
  },
]
`;
